torch/distributions/lowrank_multivariate_normal.py

```diff
@@ -73,10 +73,10 @@ class LowRankMultivariateNormal(Distribution):
 
             capacitance = I + cov_factor.T @ inv(cov_diag) @ cov_factor
     """
-    arg_constraints = {"loc": constraints.real,
-                       "cov_factor": constraints.real,
-                       "cov_diag": constraints.positive}
-    support = constraints.real
+    arg_constraints = {"loc": constraints.real_vector,
+                       "cov_factor": constraints.independent(constraints.real, 2),
+                       "cov_diag": constraints.independent(constraints.positive, 1)}
+    support = constraints.real_vector
```

Comment:
Note that the internal failures are due to checks like `if dist.support is constraints.real` or `isinstance(dist.support, constraints._Real)`. For some of the distributions, this check will need to be updated (e.g. multinomial and I don't see a way around that) in client code. Constraints wrapped within `Independent` will be harder to check this way. e.g. if we have `independent(real, 2)`, what would be the recommended way to check the base constraint without the `event dim` information?

Reply:
This points to a more general issue that I have observed, which is what is the recommended way for inferring constraint type. e.g. the most general way would be something like `isinstance(constraint, _IntegerInterval)` but that requires peeking into the non-public interface.

Reply:
@neerajprad for `constraints.independent(constraints.real, 2)` I suppose we could brute force and
```py
assert isinstance(c, constraints.independent)
assert isinstance(c.base_constraint, constraints.real)
assert isinstance(c.event_dim, 2)
```
or we could define `.__eq__()` to enable syntax like:
```py
assert c == constraints.independent(constraints.real, 2)
```
(we already do this for transforms).

Reply:
There are, I think, two issues - I created https://github.com/pytorch/pytorch/issues/50616 to discuss one of these. The other issue is checking for the constraint type (disregarding event dim and the wrapping `independent` constraint) which is where most of the usage is rather than a hard equality check. So something like, `isinstance(constraint, constraints._Real) or (isinstance(constraint, constraints._Independent) and isinstance(constraint.base_constraint, constraints._Real))` will work but only if the `independent`s aren't nested.

Reply:
I plan to use something like the following internally to handle these use cases, and we can consider providing this as a utility in PyTorch itself so as to avoid usage of non-public classes for these kind of checks. I am not sure how much usage like this we are going to find in the wild, so this may still end up breaking some code, which is my only (although minor) concern.

```python
def _unwrap(constraint):
    if isinstance(constraint, constraints.independent):
        return _unwrap(constraint.base_constraint)
    return constraint if isinstance(constraint, type) else constraint.__class__


def constraint_type_eq(constraint1, constraint2):
    return _unwrap(constraint1) == _unwrap(constraint2)
```